api/server/sdk/objectstore_test.go

```diff
@@ -0,0 +1,221 @@
+package sdk
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestObjectstoreInspectSuccess(t *testing.T) {
+	// Create server and client connection
+	s := newTestServer(t)
+	defer s.Stop()
+
+	// Create response
+	objResp := &api.ObjectstoreInfo{
+		Uuid:     "test-uuid",
+		VolumeId: "test-vol-id",
+		Enabled:  false,
+	}
+
+	s.MockCluster().
+		EXPECT().
+		ObjectStoreInspect(objResp.Uuid).
+		Return(objResp, nil)
+
+	// Setup client
+	c := api.NewOpenStorageObjectstoreClient(s.Conn())
+
+	// Get info
+	resp, err := c.InspectObjectstore(context.Background(), &api.SdkObjectstoreInspectRequest{ObjectstoreId: objResp.Uuid})
+	assert.NoError(t, err)
+	assert.NotNil(t, resp.GetObjectstoreStatus())
+
+	// Verify
+	assert.Equal(t, resp.GetObjectstoreStatus().GetVolumeId(), objResp.VolumeId)
+}
+
+func TestObjectstoreInspectFailed(t *testing.T) {
+	// Create server and client connection
+	s := newTestServer(t)
+	defer s.Stop()
+
+	// Create response
+	objResp := &api.ObjectstoreInfo{
+		Uuid:     "test-uuid",
+		VolumeId: "test-vol-id",
+		Enabled:  false,
+	}
+
+	s.MockCluster().
+		EXPECT().
+		ObjectStoreInspect(objResp.Uuid).
+		Return(nil, fmt.Errorf("some error"))
+
+	// Setup client
+	c := api.NewOpenStorageObjectstoreClient(s.Conn())
+
+	// Get info
+	resp, err := c.InspectObjectstore(context.Background(), &api.SdkObjectstoreInspectRequest{ObjectstoreId: objResp.Uuid})
+	assert.Error(t, err)
+	assert.Nil(t, resp)
+
+	// Verify
+	assert.Contains(t, err.Error(), "some error")
```

Comment:
please test error codes in the failed scenarios.
refer to the volume test cases as well as credential create/delete